networking/eth2/src/main/java/tech/pegasys/teku/networking/eth2/peers/DiscoveryNodeIdExtractor.java

```diff
@@ -13,9 +13,10 @@
 
 package tech.pegasys.teku.networking.eth2.peers;
 
+import java.util.Optional;
 import org.apache.tuweni.units.bigints.UInt256;
 import tech.pegasys.teku.networking.p2p.peer.Peer;
 
 public interface DiscoveryNodeIdExtractor {
-  UInt256 calculateDiscoveryNodeId(Peer peer);
+  Optional<UInt256> calculateDiscoveryNodeId(Peer peer);
```

Comment:
For `Libp2pPeer` the `nodeId` would always exist. We have no other `Peer` implementations aside of libp2p. Do we really need to handle hypothetical case  of absent `NodeId` ?